software/base/src/test/java/org/apache/brooklyn/entity/EC2ExpungeLiveTest.java

```diff
@@ -0,0 +1,144 @@
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import org.apache.brooklyn.api.entity.EntitySpec;
+import org.apache.brooklyn.api.location.Location;
+import org.apache.brooklyn.core.entity.Entities;
+import org.apache.brooklyn.core.internal.BrooklynProperties;
+import org.apache.brooklyn.core.location.LocationConfigKeys;
+import org.apache.brooklyn.core.location.cloud.CloudLocationConfig;
+import org.apache.brooklyn.core.mgmt.EntityManagementUtils;
+import org.apache.brooklyn.core.test.BrooklynAppLiveTestSupport;
+import org.apache.brooklyn.core.test.entity.LocalManagementContextForTests;
+import org.apache.brooklyn.core.test.entity.TestApplication;
+import org.apache.brooklyn.core.test.entity.TestEntity;
+import org.apache.brooklyn.entity.software.base.EmptySoftwareProcess;
+import org.apache.brooklyn.location.jclouds.JcloudsLocation;
+import org.apache.brooklyn.test.Asserts;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.apache.brooklyn.util.exceptions.Exceptions;
+import org.jclouds.aws.ec2.compute.AWSEC2ComputeService;
+import org.jclouds.compute.domain.ComputeMetadata;
+import org.jclouds.compute.domain.NodeMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.regex.Pattern;
+
+public class EC2ExpungeLiveTest extends BrooklynAppLiveTestSupport {
+    private static final Logger LOG = LoggerFactory.getLogger(EC2ExpungeLiveTest.class);
+
+    public static final String PROVIDER = "aws-ec2";
+    public static final String SMALL_HARDWARE_ID = "m1.small";
+
+    public static final String REGION_NAME = "us-west-2";
+    public static final String LOCATION_SPEC = PROVIDER + (REGION_NAME == null ? "" : ":" + REGION_NAME);
+
+    protected BrooklynProperties brooklynProperties;
+
+    protected Location jcloudsLocation;
+
+    @BeforeMethod(alwaysRun=true)
+    @Override
+    public void setUp() throws Exception {
+        // Don't let any defaults from brooklyn.properties (except credentials) interfere with test
+        brooklynProperties = BrooklynProperties.Factory.newDefault();
+
+        // Also removes scriptHeader (e.g. if doing `. ~/.bashrc` and `. ~/.profile`, then that can cause "stdin: is not a tty")
+        brooklynProperties.remove("brooklyn.ssh.config.scriptHeader");
+
+        mgmt = new LocalManagementContextForTests(brooklynProperties);
+        super.setUp();
+    }
+
+    /**
+     Verifies that when {@link org.apache.brooklyn.entity.software.base.lifecycle.MachineLifecycleEffectorTasks#doStop(ConfigBag, Callable<?>)} is called
+     it will not retry and wait for a MachineLocation to appear.
+     */
+    @Test
+    public void testExpungeNonJcloudsLocation() throws Exception {
```

Comment:
This test is for simple entities used in unit tests (have to improve naming here).
I am trying to add a similar Integration test for a BYON location.